CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+            ${CMAKE_INSTALL_INCLUDEDIR}/uriparser
+)
+
+#
+# C command line tool
+#
+add_executable(uriparse
+    tool/uriparse.c
+)
+
+target_link_libraries(uriparse PUBLIC uriparser)
+
+if(HAIKU)
+    # Function inet_ntop needs -lsocket or -lnetwork (see pull request #45)
+    check_library_exists(socket inet_ntop "" HAVE_LIBSOCKET__INET_NTOP)
+    check_library_exists(network inet_ntop "" HAVE_LIBNETWORK__INET_NTOP)
+    if(HAVE_LIBSOCKET__INET_NTOP)
+        target_link_libraries(uriparse PUBLIC socket)
+    endif(HAVE_LIBSOCKET__INET_NTOP)
+    if(HAVE_LIBNETWORK__INET_NTOP)
+        target_link_libraries(uriparse PUBLIC network)
+    endif(HAVE_LIBNETWORK__INET_NTOP)
+endif(HAIKU)
+
+if(WIN32)
+    target_link_libraries(uriparse PUBLIC ws2_32)
+endif(WIN32)
+
+install(
+    TARGETS
+        uriparse
+    DESTINATION
+        ${CMAKE_INSTALL_BINDIR}
+)
+
+#
+# C++ test runner
+#
+if(URIPARSER_BUILD_TESTS)
+    enable_testing()
+
+    find_package(GTest 1.8.1 REQUIRED)
+
+    add_executable(testrunner
+        test/FourSuite.cpp
+        test/MemoryManagerSuite.cpp
+        test/test.cpp
+        test/VersionSuite.cpp
+    )
+
+    target_include_directories(testrunner SYSTEM PRIVATE
+        ${GTEST_INCLUDE_DIR}
+    )
+
+    target_include_directories(testrunner PRIVATE
+        ${CMAKE_CURRENT_BINARY_DIR}  # for config.h
+    )
+
+    target_link_libraries(testrunner PUBLIC
+        uriparser
+        ${GTEST_BOTH_LIBRARIES}
+    )
+
+    add_test(
+        NAME
+            test
+        COMMAND
+            testrunner
+    )
+endif(URIPARSER_BUILD_TESTS)
+
+#
+# Doxygen API documentation
+#
+if(URIPARSER_BUILD_DOCS)
+    find_package(Doxygen REQUIRED dot doxygen)
+    find_program(QHG_LOCATION qhelpgenerator)
+
+    # Generate Doxyfile
+    set(PACKAGE ${PROJECT_NAME})
+    set(VERSION ${PROJECT_VERSION})
+    set(ac_abs_confdir ${CMAKE_CURRENT_SOURCE_DIR})
+    set(GENERATE_HTMLHELP NO)
+    configure_file(doc/Doxyfile.in doc/Doxyfile @ONLY)
+    configure_file(doc/release.sh.in doc/release.sh @ONLY)
+
+    add_custom_target(doc
+        COMMAND
+            ${DOXYGEN_EXECUTABLE}
+            Doxyfile
+        WORKING_DIRECTORY
+            ${CMAKE_CURRENT_BINARY_DIR}/doc
+        COMMENT
+            "Generating API documentation with Doxygen"
+        VERBATIM
+    )
+    add_dependencies(uriparser doc)
+
+    install(
+        DIRECTORY
+            ${CMAKE_CURRENT_BINARY_DIR}/doc/html
+        DESTINATION
+            ${CMAKE_INSTALL_DOCDIR}
+    )
+    if(QHG_LOCATION)
+        install(
+            FILES
+                ${CMAKE_CURRENT_BINARY_DIR}/doc/uriparser-${PROJECT_VERSION}.qch
+            DESTINATION
+                ${CMAKE_INSTALL_DOCDIR}
+        )
+    endif(QHG_LOCATION)
+endif(URIPARSER_BUILD_DOCS)
+
+#
+# CMake files for find_package(uriparser [..])
+#
+install(
+    EXPORT
+        uriparser
+    DESTINATION
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser
+    NAMESPACE
+        uriparser::
+)
```

Comment:
You need a bit more.

Example:

https://github.com/markand/uriparser/blob/master/CMakeLists.txt#L140-L159
https://github.com/markand/uriparser/blob/master/cmake/uriparser-config.cmake.in

And specify EXPORT parameter in all targets.

https://github.com/markand/uriparser/blob/master/CMakeLists.txt#L102-L108

A full example is available on CMake about [packages](https://cmake.org/cmake/help/v3.13/manual/cmake-packages.7.html#creating-packages) but not the whole thing is necessary.

Yes, unfortunately creating packages in CMake is not the easiest thing.

Reply:
Wow, not exactly trivial. Do you have a few minutes to review the new `find_package` bytes?

Related commit: c420a839e547119d3749e1b44a06f7fd8791c63d